crates/fluvio/src/producer/config.rs

```diff
@@ -0,0 +1,67 @@
+use std::time::Duration;
+
+use crate::producer::partitioning::{Partitioner, SiphashRoundRobinPartitioner};
+
+const DEFAULT_LINGER_MS: u64 = 250;
+const DEFAULT_BATCH_SIZE_BYTES: usize = 16_000;
+
+/// Builder for `TopicProducerConfig`
+#[derive(Default)]
```

Comment:
Can you use DeriveBuilder?

Reply:
I think that in order to use that we need to make ` partitioner: Box<dyn Partitioner + Send + Sync>` clonable

Reply:
Can use `#[builder(pattern = "owned")] ` which doesn't require cloning